Address some feedback from chapter 12 (#984)
browserengineering/book: book/scheduling.md

        self.tab.set_needs_render()
```

> There are more calls to `render`; you should find and fix all of them...except
> `click`. Here's why:

Reviewer: ```suggestion
There are more calls to `render`; you should find and fix all of them...
Except, let's take a closer look at `click`.
```

> Now we don't immediately render when something changes. That means that the

Reviewer: ```suggestion
We now don't immediately render when something changes. That means that the
```

```

There are more calls to `render`; you should find and fix all of them.
However, there is one call to `render` at the beginning of `click` that
should remain, because it's necessary to make sure that layout is up-to-date

> before finding which layout object is under the click.

Reviewer: I think this still isn't right. I went back through and Ch 9 is the last one that touches `click` (it adds JS event dispatching). But Ch9 doesn't render at the top of `click`—because in Chapter 9 we eagerly render any time we change, so there's no need to do it at the top of `click`. I think this paragraph should be something like:

> Now we don't immediately render on change. That means that the layout tree (and style) could be out of date when a method is called. Normally, this isn't a problem, but in one important case it is: click handling. That's because we need to *read* the layout tree to figure out what object was clicked on, which means the layout tree needs to be up to date. Add a call to `render` at the top of `click`:
> ... code block here

When running an animation frame, the `Tab` should construct one of
these objects and pass it to `commit`. To keep `render` from getting
too confusing, let's put this in a new `run_animation_frame` method,
and move `__runRAFHandlers` there too.^[I'm not just adding all this
to `render`, because `render` is just about updating style, layout and paint
when needed. That's why it is called from the beginning `click` instead of
`run_animation_frame`. The latter also has script callbacks and also commits

> to the browser, which don't make sense to have happen from `click`.]

Reviewer: This is a fantastic reason.[1]

[1] Unrelated, but I found it funny that we made the right architecture decision to account for a control flow path that our browser, due to a bug, actually didn't have!

Reviewer: ```suggestion
and move `__runRAFHandlers` there too.^[The `render` method is just about
updating style, layout and paint when needed; it's called for every frame,
but it's also called from `click`, and in real browsers from many other places too.
Meanwhile, `run_animation_frame` is only called for frames, and therefore
it, not `render`, runs RAF handlers and calls `commit`.]
```

Rephrase.
